### DISCUSSION TITLE
Feature: Javascript SDK and WebPush support

### DISCUSSION THREAD

**Discussion by parvardegr** [parcelvoy/platform#488] [Ideas]
Hello,
Do we have any plan to implement them in the near future? maybe i can help with them too!

**Comment by pushchris** [comment]
Currently no plans for a Javascript SDK, though it wouldn't be crazy hard to create (you can reference the docs for what is implemented client side here https://docs.parcelvoy.com/api/client)

As for WebPush, this is the first request we've had for it! Technically the[ package we are using for sending notifications](https://github.com/appfeel/node-pushnotifications?tab=readme-ov-file#web-push) supports it but would take some finagling to properly support all of the use cases.

You are welcome to make contributions to either!

**Comment by pushchris** [comment]
There is now a beta JS SDK you are welcome to try out. It is definitely super early so please let us know if you run into any issues with it: https://github.com/parcelvoy/js-sdk

**Comment by teawithfruit** [comment]
Thanks for the JS Library.

I still have a question regarding the use of ˋclient.track({})ˋ. Is it possible to get live in-app updates by using this function like it would be possible with websockets?

Is it generally possible or is it planned to make in-app notifications possible, if not already possible? In other words, that I can send messages to a user on a self-hosted website and use something like a notification center?

**Reply by pushchris** [reply]
For mobile apps you could listen for push notifications and respond accordingly, but there is no support for silent notifications at this time so the user would get alerted. But it would let you trigger code to run on your end.

In-app messaging is in the works, but web-support is not planned for initial release.

If you are primarily looking for notifications, a product like https://novu.co might fit your use case better.
